File: backend/src/main/java/com/plugga/backend/entity/Side.java
package com.plugga.backend.entity;

import javax.persistence.*;

@Entity
@Table(name="side")
public class Side {

    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    @Column(name="side_id")
    private int id;

    @ManyToOne(cascade={CascadeType.DETACH, CascadeType.MERGE, CascadeType.PERSIST, CascadeType.REFRESH})
    @JoinColumn(name="card_id")
    private int card;

    @Column(name="image_url")
    private String imageUrl;

    @Column(name="text")
    private String text;

    public Side() {
    }

    public Side(final String imageUrl, final String text) {
        this.imageUrl = imageUrl;
        this.text = text;
    }

    public int getId() {
        return id;
    }

    public void setId(final int id) {
        this.id = id;
    }

    public int getCard() {
        return card;
    }

    public void setCard(final int card) {
        this.card = card;
    }

    public String getImageUrl() {
        return imageUrl;
    }

    public void setImageUrl(final String imageUrl) {
        this.imageUrl = imageUrl;
    }

    public String getText() {
        return text;
    }

    public void setText(final String text) {
        this.text = text;
    }

    @Override
    public String toString() {
        return "Side{" +
                "id=" + id +
                ", card=" + card +
                ", imageUrl='" + imageUrl + '\'' +
                ", text='" + text + '\'' +
                '}';
    }
}

File: backend/src/main/kotlin/com/plugga/backend/entity/Side.kt
package com.plugga.backend.entity

import com.fasterxml.jackson.annotation.JsonIgnore

import javax.persistence.*

@Entity
@Table(name = "side")
class Side {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "side_id")
    var id: Int = 0

    @ManyToOne
    @JoinColumn(name = "card_id", referencedColumnName = "card_id")
    @get:JsonIgnore
    var card: Card? = null

    @Column(name = "image_url")
    var imageUrl: String? = null

    @Column(name = "text")
    var text: String? = null

    constructor() {}

    constructor(id: Int) {
        this.id = id
    }

    constructor(imageUrl: String, text: String) {
        this.imageUrl = imageUrl
        this.text = text
    }

    override fun toString(): String {
        return "Side{" +
                "id=" + id +
                ", card=" + card +
                ", imageUrl='" + imageUrl + '\''.toString() +
                ", text='" + text + '\''.toString() +
                '}'.toString()
    }
}
